java/src/test/java/linalg/algorithms/GramSchmidtTest.java

```diff
@@ -104,37 +121,32 @@ void DecomposeVector() {
       { 1.0, 2.0, 3.0 }
     };
 
-    double[][] arr_q = {
-      { 1.0, 0.0, 0.0 }
-    };
-
-    double[][] arr_r = {
-      { 1.0, 2.0, 3.0 },
-      { 0.0, 0.0, 0.0 },
-      { 0.0, 0.0, 0.0 }
-    };
-
     DenseMatrixBuilder dmb = new DenseMatrixBuilder(1, 3);
     for (int i = 0; i < 3; i++)
-        dmb.SetValue(0, i, arr[0][i]);
+      dmb.SetValue(0, i, arr[0][i]);
 
-    GramSchmidt res = new GramSchmidt(dmb.BuildMatrix());
-    res.Decompose();
-    Matrix Q = res.Q();
-    Matrix R = res.R();
+    gram_schmidt.Decompose(dmb.BuildMatrix());
 
-    Assertions.assertEquals(arr_r.length, R.Rows());
-    Assertions.assertEquals(arr_r[0].length, R.Cols());
+    Matrix Q = gram_schmidt.GetQ();
+    Matrix R = gram_schmidt.GetR();
 
-    Assertions.assertEquals(arr_q.length, Q.Rows());
-    Assertions.assertEquals(arr_q[0].length, Q.Cols());
+    Matrix id = MatrixFactory.DenseMultiply(Q, Q.Transpose());
+    Matrix res = MatrixFactory.DenseMultiply(Q, R);
 
-    for (int i = 0; i < Q.Rows(); i++)
-      for (int j = 0; j < Q.Cols(); j++)
-        Assertions.assertEquals(arr_q[i][j], Q.ValueAt(i, j));
+    Assertions.assertEquals(id.Rows(), id.Cols());
+    Assertions.assertEquals(arr.length, res.Rows());
+    Assertions.assertEquals(arr[0].length, res.Cols());
 
-    for (int i = 0; i < R.Rows(); i++)
-      for (int j = 0; j < R.Cols(); j++)
-        Assertions.assertEquals(arr_r[i][j], R.ValueAt(i, j));
+    for (int i = 0; i < id.Rows(); i++)
```

Comment:
Same here, you can see that having a separate function would make life a lot easier